syft/frameworks/torch/mpc/aby3/aby3_helper.py

```diff
@@ -0,0 +1,70 @@
+import torch
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+from syft.frameworks.torch.mpc.przs import gen_alpha_3of3
+
+class ABY3Helper:
+    def bit_inject(rst_binary):
+        assert rst_binary.ring_size == 2
+
+        return_wrapper = rst_binary.is_wrapper
+        if return_wrapper:
+            rst_binary = rst_binary.child
+
+        players = rst_binary.players
+        shares_p1, shares_p2, shares_p3 = rst_binary.retrieve_pointers()
+
+        """
+            x shared in ring size 2
+            x_b = (x_b_1, x_b_2, x_b_3)
+
+            Share the shares in ring size L
+            Resharig the binary shars in arithmetic ones should have no communication cost
+            but because we use an orchestrator there is added the communicatio overhead
+
+            x1_shares = ((x_b_1, 0), (0, 0), (0, x_b_1))
+            x2_shares = ((0, x_b_2), (x_b_2, 0), (0, 0))
+            x3_shares = ((0, 0), (0, x_b_3), (x_b_3, 0))
+
+            For emulating a xor on arithmetic shares:
+                * [[v1 xor v2]] = [[v1]] + [[v2]] - 2 * [[v1]] * [[v2]]
+
+            (x1_shares xor x2_shares xor x3_shares) == x_b (ring size L)
+        """
+
+        zero_share_p1 = gen_alpha_3of3(players[0]).wrap()
+        zero_share_p2 = gen_alpha_3of3(players[1]).wrap()
+        zero_share_p3 = gen_alpha_3of3(players[2]).wrap()
+
+        """
+        Since we shared the seeds at the beginning we know that
+        zero_share_p1 + zero_share_p2 + zero_share_p3 == 0
+        """
+
+        b1_share_p1 = (shares_p1[0] + zero_share_p1, zero_share_p2.copy().move(players[0]))
+        b1_share_p2 = (zero_share_p2, zero_share_p3.copy().move(players[1]))
+        b1_share_p3 = (zero_share_p3, shares_p3[1] + zero_share_p1.copy().move(players[2]))
+
+        b2_share_p1 = (zero_share_p1.copy(), shares_p1[1] + zero_share_p2.copy().move(players[0]))
```

Comment:
not sure you need the copy() here:
`(zero_share_p1.copy(), sh`  -> `(zero_share_p1, sh`

Reply:
ah maybe not... no you're right